From 2683419e05ab227cd24a5859db044fe3941d6574 Mon Sep 17 00:00:00 2001
From: Dave Jones <davej@redhat.com>
Date: Mon, 13 May 2013 16:53:02 -0400
Subject: [PATCH 11/44] silence fuzz-testing noise

diff --git a/arch/x86/ia32/sys_ia32.c b/arch/x86/ia32/sys_ia32.c
index 8e0ceecdc957..e8071976793c 100644
--- a/arch/x86/ia32/sys_ia32.c
+++ b/arch/x86/ia32/sys_ia32.c
@@ -203,15 +203,6 @@ long sys32_fadvise64_64(int fd, __u32 offset_low, __u32 offset_high,
 
 long sys32_vm86_warning(void)
 {
-	struct task_struct *me = current;
-	static char lastcomm[sizeof(me->comm)];
-
-	if (strncmp(lastcomm, me->comm, sizeof(lastcomm))) {
-		compat_printk(KERN_INFO
-			      "%s: vm86 mode not supported on 64 bit kernel\n",
-			      me->comm);
-		strncpy(lastcomm, me->comm, sizeof(lastcomm));
-	}
 	return -ENOSYS;
 }
 
diff --git a/arch/x86/kernel/vm86_32.c b/arch/x86/kernel/vm86_32.c
index e8edcf52e069..95147acf071c 100644
--- a/arch/x86/kernel/vm86_32.c
+++ b/arch/x86/kernel/vm86_32.c
@@ -138,17 +138,14 @@ struct pt_regs *save_v86_state(struct kernel_vm86_regs *regs)
 	 */
 	local_irq_enable();
 
-	if (!current->thread.vm86_info) {
-		pr_alert("no vm86_info: BAD\n");
+	if (!current->thread.vm86_info)
 		do_exit(SIGSEGV);
-	}
+
 	set_flags(regs->pt.flags, VEFLAGS, X86_EFLAGS_VIF | current->thread.v86mask);
 	tmp = copy_vm86_regs_to_user(&current->thread.vm86_info->regs, regs);
 	tmp += put_user(current->thread.screen_bitmap, &current->thread.vm86_info->screen_bitmap);
-	if (tmp) {
-		pr_alert("could not access userspace vm86_info\n");
+	if (tmp)
 		do_exit(SIGSEGV);
-	}
 
 	tss = &per_cpu(init_tss, get_cpu());
 	current->thread.sp0 = current->thread.saved_sp0;
diff --git a/drivers/gpu/drm/drm_sysfs.c b/drivers/gpu/drm/drm_sysfs.c
index cc3d6d6d67e0..306b3a9189b9 100644
--- a/drivers/gpu/drm/drm_sysfs.c
+++ b/drivers/gpu/drm/drm_sysfs.c
@@ -285,7 +285,6 @@ static ssize_t subconnector_show(struct device *device,
 	}
 
 	if (!prop) {
-		DRM_ERROR("Unable to find subconnector property\n");
 		return 0;
 	}
 
@@ -326,7 +325,6 @@ static ssize_t select_subconnector_show(struct device *device,
 	}
 
 	if (!prop) {
-		DRM_ERROR("Unable to find select subconnector property\n");
 		return 0;
 	}
 
diff --git a/drivers/scsi/scsi_netlink.c b/drivers/scsi/scsi_netlink.c
index 109802f776ed..1e8b706967fa 100644
--- a/drivers/scsi/scsi_netlink.c
+++ b/drivers/scsi/scsi_netlink.c
@@ -55,11 +55,8 @@ scsi_nl_rcv_msg(struct sk_buff *skb)
 
 		nlh = nlmsg_hdr(skb);
 		if ((nlh->nlmsg_len < (sizeof(*nlh) + sizeof(*hdr))) ||
-		    (skb->len < nlh->nlmsg_len)) {
-			printk(KERN_WARNING "%s: discarding partial skb\n",
-				 __func__);
+		    (skb->len < nlh->nlmsg_len))
 			return;
-		}
 
 		rlen = NLMSG_ALIGN(nlh->nlmsg_len);
 		if (rlen > skb->len)
@@ -82,11 +79,8 @@ scsi_nl_rcv_msg(struct sk_buff *skb)
 			goto next_msg;
 		}
 
-		if (nlh->nlmsg_len < (sizeof(*nlh) + hdr->msglen)) {
-			printk(KERN_WARNING "%s: discarding partial message\n",
-				 __func__);
+		if (nlh->nlmsg_len < (sizeof(*nlh) + hdr->msglen))
 			goto next_msg;
-		}
 
 		/*
 		 * Deliver message to the appropriate transport
diff --git a/kernel/signal.c b/kernel/signal.c
index 16a305295256..e1ebea3db6e7 100644
--- a/kernel/signal.c
+++ b/kernel/signal.c
@@ -3044,7 +3044,6 @@ static int do_rt_tgsigqueueinfo(pid_t tgid, pid_t pid, int sig, siginfo_t *info)
 	if (((info->si_code >= 0 || info->si_code == SI_TKILL)) &&
 	    (task_pid_vnr(current) != pid)) {
 		/* We used to allow any < 0 si_code */
-		WARN_ON_ONCE(info->si_code < 0);
 		return -EPERM;
 	}
 	info->si_signo = sig;
diff --git a/net/appletalk/ddp.c b/net/appletalk/ddp.c
index 0d0766ea5ab1..4362a4404078 100644
--- a/net/appletalk/ddp.c
+++ b/net/appletalk/ddp.c
@@ -1204,12 +1204,7 @@ static int atalk_connect(struct socket *sock, struct sockaddr *uaddr,
 
 	if (addr->sat_addr.s_node == ATADDR_BCAST &&
 	    !sock_flag(sk, SOCK_BROADCAST)) {
-#if 1
-		pr_warn("atalk_connect: %s is broken and did not set SO_BROADCAST.\n",
-			current->comm);
-#else
 		return -EACCES;
-#endif
 	}
 
 	lock_sock(sk);
diff --git a/net/can/af_can.c b/net/can/af_can.c
index 66e08040ced7..8cd31fd8ff8b 100644
--- a/net/can/af_can.c
+++ b/net/can/af_can.c
@@ -154,15 +154,6 @@ static int can_create(struct net *net, struct socket *sock, int protocol,
 
 		err = request_module("can-proto-%d", protocol);
 
-		/*
-		 * In case of error we only print a message but don't
-		 * return the error code immediately.  Below we will
-		 * return -EPROTONOSUPPORT
-		 */
-		if (err)
-			printk_ratelimited(KERN_ERR "can: request_module "
-			       "(can-proto-%d) failed.\n", protocol);
-
 		cp = can_get_proto(protocol);
 	}
 #endif
diff --git a/net/core/sock.c b/net/core/sock.c
index 1c7a33db1314..b981f9108eb5 100644
--- a/net/core/sock.c
+++ b/net/core/sock.c
@@ -392,14 +392,7 @@ static int sock_set_timeout(long *timeo_p, char __user *optval, int optlen)
 		return -EDOM;
 
 	if (tv.tv_sec < 0) {
-		static int warned __read_mostly;
-
 		*timeo_p = 0;
-		if (warned < 10 && net_ratelimit()) {
-			warned++;
-			pr_info("%s: `%s' (pid %d) tries to set negative timeout\n",
-				__func__, current->comm, task_pid_nr(current));
-		}
 		return 0;
 	}
 	*timeo_p = MAX_SCHEDULE_TIMEOUT;
diff --git a/net/ipv4/raw.c b/net/ipv4/raw.c
index 0bb68df5055d..65772ffdde12 100644
--- a/net/ipv4/raw.c
+++ b/net/ipv4/raw.c
@@ -518,8 +518,6 @@ static int raw_sendmsg(struct kiocb *iocb, struct sock *sk, struct msghdr *msg,
 		if (msg->msg_namelen < sizeof(*usin))
 			goto out;
 		if (usin->sin_family != AF_INET) {
-			pr_info_once("%s: %s forgot to set AF_INET. Fix it!\n",
-				     __func__, current->comm);
 			err = -EAFNOSUPPORT;
 			if (usin->sin_family)
 				goto out;
diff --git a/net/irda/af_irda.c b/net/irda/af_irda.c
index 568edc72d737..fa19e6ef5ff3 100644
--- a/net/irda/af_irda.c
+++ b/net/irda/af_irda.c
@@ -494,10 +494,8 @@ static int irda_open_lsap(struct irda_sock *self, int pid)
 {
 	notify_t notify;
 
-	if (self->lsap) {
-		net_warn_ratelimited("%s(), busy!\n", __func__);
+	if (self->lsap)
 		return -EBUSY;
-	}
 
 	/* Initialize callbacks to be used by the IrDA stack */
 	irda_notify_init(&notify);
@@ -2105,8 +2103,6 @@ static int irda_setsockopt(struct socket *sock, int level, int optname,
 				 __func__, opt);
 			self->max_sdu_size_rx = opt;
 		} else {
-			net_warn_ratelimited("%s: not allowed to set MAXSDUSIZE for this socket type!\n",
-					     __func__);
 			err = -ENOPROTOOPT;
 			goto out;
 		}
diff --git a/net/netfilter/ipvs/ip_vs_core.c b/net/netfilter/ipvs/ip_vs_core.c
index 990decba1fe4..ce9887ca8109 100644
--- a/net/netfilter/ipvs/ip_vs_core.c
+++ b/net/netfilter/ipvs/ip_vs_core.c
@@ -1975,8 +1975,6 @@ static int __net_init __ip_vs_init(struct net *net)
 	if (ip_vs_sync_net_init(net) < 0)
 		goto sync_fail;
 
-	printk(KERN_INFO "IPVS: Creating netns size=%zu id=%d\n",
-			 sizeof(struct netns_ipvs), ipvs->gen);
 	return 0;
 /*
  * Error handling
diff --git a/net/rds/bind.c b/net/rds/bind.c
index a2e6562da751..6294c5b9f259 100644
--- a/net/rds/bind.c
+++ b/net/rds/bind.c
@@ -185,8 +185,6 @@ int rds_bind(struct socket *sock, struct sockaddr *uaddr, int addr_len)
 	if (!trans) {
 		ret = -EADDRNOTAVAIL;
 		rds_remove_bound(rs);
-		printk_ratelimited(KERN_INFO "RDS: rds_bind() could not find a transport, "
-				"load rds_tcp or rds_rdma?\n");
 		goto out;
 	}
 
diff --git a/net/sctp/socket.c b/net/sctp/socket.c
index 2625eccb77d5..2678af5c129d 100644
--- a/net/sctp/socket.c
+++ b/net/sctp/socket.c
@@ -6312,8 +6312,6 @@ static int sctp_listen_start(struct sock *sk, int backlog)
 		sprintf(alg, "hmac(%s)", sp->sctp_hmac_alg);
 		tfm = crypto_alloc_hash(alg, 0, CRYPTO_ALG_ASYNC);
 		if (IS_ERR(tfm)) {
-			net_info_ratelimited("failed to load transform for %s: %ld\n",
-					     sp->sctp_hmac_alg, PTR_ERR(tfm));
 			return -ENOSYS;
 		}
 		sctp_sk(sk)->hmac = tfm;
diff --git a/net/socket.c b/net/socket.c
index a2c33a4dc7ba..326e4a9ebcd9 100644
--- a/net/socket.c
+++ b/net/socket.c
@@ -2592,8 +2592,6 @@ int sock_register(const struct net_proto_family *ops)
 		err = 0;
 	}
 	spin_unlock(&net_family_lock);
-
-	pr_info("NET: Registered protocol family %d\n", ops->family);
 	return err;
 }
 EXPORT_SYMBOL(sock_register);
diff --git a/security/selinux/hooks.c b/security/selinux/hooks.c
index 6da7532893a1..d4588654ee68 100644
--- a/security/selinux/hooks.c
+++ b/security/selinux/hooks.c
@@ -4723,10 +4723,6 @@ static int selinux_nlmsg_perm(struct sock *sk, struct sk_buff *skb)
 	err = selinux_nlmsg_lookup(sksec->sclass, nlh->nlmsg_type, &perm);
 	if (err) {
 		if (err == -EINVAL) {
-			printk(KERN_WARNING
-			       "SELinux: unrecognized netlink message:"
-			       " protocol=%hu nlmsg_type=%hu sclass=%hu\n",
-			       sk->sk_protocol, nlh->nlmsg_type, sksec->sclass);
 			if (!selinux_enforcing || security_get_allow_unknown())
 				err = 0;
 		}
diff --git a/sound/pci/hda/hda_codec.c b/sound/pci/hda/hda_codec.c
index 2fe86d2e1b09..7643b0e7c6da 100644
--- a/sound/pci/hda/hda_codec.c
+++ b/sound/pci/hda/hda_codec.c
@@ -581,12 +581,10 @@ int snd_hda_get_raw_connections(struct hda_codec *codec, hda_nid_t nid,
 		}
 		range_val = !!(parm & (1 << (shift-1))); /* ranges */
 		val = parm & mask;
-		if (val == 0 && null_count++) {  /* no second chance */
-			codec_dbg(codec,
-				  "invalid CONNECT_LIST verb %x[%i]:%x\n",
-				    nid, i, parm);
+
+		if (val == 0 && null_count++)	/* no second chance */
 			return 0;
-		}
+
 		parm >>= shift;
 		if (range_val) {
 			/* ranges between the previous and this one */
